[wip] Implement new findings views

jansorg opened the pull request:
Closes https://github.com/getappmap/appmap-intellij-plugin/issues/174

dustinbyrne commented:
The commits look good. I still need to give the plugin a run through.

ahtrotta commented:
I ran `./gradlew runIde` and tested the extension and it looks good! I only have two small comments that aren't blocking. These could be follow-on items.

1. It would be nice if the titles of the Finding Details webviews weren't all `Finding Details` so users could quickly see what findings were open in the editor:

![image](https://user-images.githubusercontent.com/45714532/208758857-1878cb78-8e6a-4e75-90db-b2e2cbf8916e.png)


Here's how it looks in VS Code:

![image](https://user-images.githubusercontent.com/45714532/208759347-6fd74951-692e-425b-a98a-737c46cad4c9.png)


2. Also, it would be better if the Findings Details webviews were re-used, so that a new one wasn't opened every time, even for the same finding.

dustinbyrne commented:
Can we add the following telemetry as well?

**analysis:view_overview**
```ts
{
  "event": "appland.appmap/analysis:view_overview",
  "properties": {
    "appmap.analysis.rules": "unbatched-materialized-query,secret-in-log", // Comma delimited list of unique rule identifiers
    "appmap.analysis.impact_domains": "performance,security" // Comma delimited list of unique impact domains
  },
  "metrics": {
    "num_findings.num_total_findings": 3, // The total number of non-unique findings
    "num_findings.num_unique_findings": 1, // The number of findings after de-duplication by hash
    "num_findings.num_performance": 2, // The total number of findings split by domain
    "num_findings.num_security": 1,
    "num_findings.num_unique_performance": 1, // The number of non-unique findings after deduplication by hash split by impact domain
    "num_findings.num_unique_security": 1
  }
}
```
In the metrics in the above example, `num_findings.num_${impact_domain}` and `num_findings.num_unique_${impact_domain}` exist for every lower case value of `impact_domain` (`security`, `performance`, `maintainability`, `stability`) but these values can be omitted if the count is zero.
 

**analysis:view_finding**
```ts
{
  "event": "appland.appmap/analysis:view_finding",
  "properties": {
    "appmap.finding.rule": "unbatched-materialized-query", // The rule identifier of the finding viewed
    "appmap.finding.impact_domain": "performance", // The impact domain of the finding viewed
    "appmap.finding.hash": "a806b8da7d6fa5e1d5294ee0f3a63a715a2a6962d2af8ad5628a72baf26ca2f0" // HashV2 or HashV1
  }
}
```

jansorg commented:
> 1. It would be nice if the titles of the Finding Details webviews weren't all `Finding Details` so users could quickly see what findings were open in the editor:

Fixed by https://github.com/getappmap/appmap-intellij-plugin/pull/175/commits/932995a884094f8713dd0bea6f415e3d8643fe1f, it's now showing the title of the findings category (same as VSCode, I believe).
 
> 2. Also, it would be better if the Findings Details webviews were re-used, so that a new one wasn't opened every time, even for the same finding.

Fixed by https://github.com/getappmap/appmap-intellij-plugin/pull/175/commits/6535eb5a6e17681d1705e9b2a95f4044e429c930. An existing editor is now reused for the same finding. A new editor is created for a different findings (same as in VSCode, I think).

jansorg commented:
@dustinbyrne Adding the new messages should be possible.
I'm assuming that the messages should be sent when an editor is created. 
Should it be sent only for new editors or also when an editor was reused (overview and details webviews)?

jansorg commented:
@dustinbyrne 
581c21f70a6cbf851c7fbbe42f2e73a0ea9d4c2e adds telemetry for the finding details view
ccf051d1f922e438aa2a6d47f6faa8693ec8b01a adds telemetry for the findings overview view. 
I've reviewed the sent data manually, there's currently no test covering the telemetry data.

ahtrotta reviewed (approved):
Looks good to me, but I didn't review the code, I just reviewed the functionality.

appland-release commented:
:tada: This PR is included in version 0.23.0 :tada:

The release is available on:
- `v0.23.0`
- [GitHub release](https://github.com/getappmap/appmap-intellij-plugin/releases/tag/v0.23.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
